ratbagd/ratbagd-profile.c

```diff
@@ -509,8 +509,6 @@ ratbagd_profile_set_report_rate(sd_bus *bus,
 	/* basic sanity check */
 	if (rate > 5000)
 		rate = 5000;
-	else if (rate % 100)
-		rate = rate - (rate % 100);
 
```

Comment:
tbh, I don't recommend this change. There are very few devices that need a report rate of e.g. 476Hz and where this happens its likely more a bug than anything. The issue from a quick skim is that you need 125Hz and 250Hz to work - just add those explicitly to the list here and keep the alignment. I doubt even rates like 2100 Hz are useful - my guess is 125, 250, 500, 1000 and from then on multiples of 1000.

Reply:
I've breezed through all the drivers, and most supported report rates are 125, 250, 500 and 1000 hz. There were also several (or just one, don't remember) mice with 333 and 750 hz added to the list. Logitech G600 driver also defines support for 142, 166 and 200 hz. Checking for all of them would be cumbersome. If we do such such a check after all, I think we should align to 125 because it scales to all the most popular rates, though it won't work with these more rare report rates.
Although IMHO such check should be done on driver level. If there are devices that support such random report rates, why limit them?

Reply:
Tbh I don't see why limiting the report rates lower than the max makes any sense for normal use. In https://github.com/libratbag/piper/issues/491 the only reason is some issues with wine. So IMHO 125Hz (apparently HID default), 250Hz (for wine), 500Hz (max for older gaming mice), 1000Hz (normal mice) and multiple of 1000 (over-the-top mice) should be enough.

Reply:
@stephanlachnit, I didn't quite catch that. Do you mean we should only allow 125, 250, 500 and the multiples of 1000, or we should not do the check at all?